### ISSUE TITLE
Problem with Database Table Types for foreign keys between point_group_contact_score leads and point_groups?

### ISSUE THREAD

**Issue by technik-kvz** [mautic/mautic#13282]
### Mautic Version

5.0.x series

### PHP version

8.1

### What browsers are you seeing the problem on?

Not relevant

### What happened?

See _Originally posted by @christopheg in https://github.com/mautic/mautic/issues/12993#issuecomment-1908852656_  and following comments.

### How can we reproduce this issue?

e.g. delete table point_group_contact_score  and run migrations - that will create point_group_contact_score  and try to create a foreign key to leads - pointing from an unsigned int to a signed int.

### Relevant log output

```shell
Site works, but migrations cannot be finished:

Migrations with migrate:

php -d memory_limit=4096M bin/console doctrine:migration:migrate --no-interaction
[2024-01-23 08:22:15]
[2024-01-23 08:22:15]
[2024-01-23 08:22:15] [WARNING] You have 183 previously executed migrations in the database that are not registered migrations.
[2024-01-23 08:22:15]
[2024-01-23 08:22:15]
[2024-01-23 08:22:15] >> (20150402000000)
[2024-01-23 08:22:15] >> (20150504000000)
[2024-01-23 08:22:15] >> (20150521000000)
[2024-01-23 08:22:15] >> (20150718000000)
[2024-01-23 08:22:15] >> (20150724000000)
[2024-01-23 08:22:15] >> (20150801000000)
[2024-01-23 08:22:15] >> (20150829000000)
[2024-01-23 08:22:15] >> (20150901000000)
[2024-01-23 08:22:15] >> (20151022000000)
[2024-01-23 08:22:15] >> (20151120000000)
[2024-01-23 08:22:15] >> (20151207000000)
[2024-01-23 08:22:15] >> (20160114000000)
[2024-01-23 08:22:15] >> (20160225000000)
[2024-01-23 08:22:15] >> (20160405000000)
[2024-01-23 08:22:15] >> (20160414000000)
[2024-01-23 08:22:15] >> (20160420000000)
[2024-01-23 08:22:15] >> (20160426000000)
[2024-01-23 08:22:15] >> (20160429000000)
[2024-01-23 08:22:15] >> (20160504000000)
[2024-01-23 08:22:15] >> (20160506000000)
[2024-01-23 08:22:15] >> (20160520000000)
[2024-01-23 08:22:15] >> (20160523000000)
[2024-01-23 08:22:15] >> (20160606000000)
[2024-01-23 08:22:15] >> (20160615000000)
[2024-01-23 08:22:15] >> (20160617000318)
[2024-01-23 08:22:15] >> (20160624032452)
[2024-01-23 08:22:15] >> (20160630000000)
[2024-01-23 08:22:15] >> (20160630000001)
[2024-01-23 08:22:15] >> (20160630000002)
[2024-01-23 08:22:15] >> (20160712000000)
[2024-01-23 08:22:15] >> (20160712000001)
[2024-01-23 08:22:15] >> (20160719000000)
[2024-01-23 08:22:15] >> (20160720000000)
[2024-01-23 08:22:15] >> (20160722000000)
[2024-01-23 08:22:15] >> (20160725161822)
[2024-01-23 08:22:15] >> (20160726000000)
[2024-01-23 08:22:15] >> (20160726000001)
[2024-01-23 08:22:15] >> (20160728000000)
[2024-01-23 08:22:15] >> (20160731000000)
[2024-01-23 08:22:15] >> (20160805000123)
[2024-01-23 08:22:15] >> (20160808000000)
[2024-01-23 08:22:15] >> (20160916000000)
[2024-01-23 08:22:15] >> (20160919204648)
[2024-01-23 08:22:15] >> (20160920195943)
[2024-01-23 08:22:15] >> (20160926000000)
[2024-01-23 08:22:15] >> (20160926000001)
[2024-01-23 08:22:15] >> (20160926182807)
[2024-01-23 08:22:15] >> (20161004080958)
[2024-01-23 08:22:15] >> (20161004090629)
[2024-01-23 08:22:15] >> (20161004123446)
[2024-01-23 08:22:15] >> (20161024162029)
[2024-01-23 08:22:15] >> (20161026110456)
[2024-01-23 08:22:15] >> (20161026202839)
[2024-01-23 08:22:15] >> (20161031134707)
[2024-01-23 08:22:15] >> (20161116195435)
[2024-01-23 08:22:15] >> (20161122172519)
[2024-01-23 08:22:15] >> (20161122215214)
[2024-01-23 08:22:15] >> (20161123225456)
[2024-01-23 08:22:15] >> (20161124145649)
[2024-01-23 08:22:15] >> (20161125002837)
[2024-01-23 08:22:15] >> (20161222183556)
[2024-01-23 08:22:15] >> (20170106102310)
[2024-01-23 08:22:15] >> (20170108012944)
[2024-01-23 08:22:15] >> (20170109025947)
[2024-01-23 08:22:15] >> (20170113015255)
[2024-01-23 08:22:15] >> (20170113143922)
[2024-01-23 08:22:15] >> (20170127205928)
[2024-01-23 08:22:15] >> (20170216221648)
[2024-01-23 08:22:15] >> (20170303000000)
[2024-01-23 08:22:15] >> (20170323111702)
[2024-01-23 08:22:15] >> (20170324112219)
[2024-01-23 08:22:15] >> (20170330165232)
[2024-01-23 08:22:15] >> (20170404173451)
[2024-01-23 08:22:15] >> (20170405194627)
[2024-01-23 08:22:15] >> (20170410210055)
[2024-01-23 08:22:15] >> (20170420122443)
[2024-01-23 08:22:15] >> (20170429092049)
[2024-01-23 08:22:15] >> (20170503143650)
[2024-01-23 08:22:15] >> (20170506144649)
[2024-01-23 08:22:15] >> (20170515222314)
[2024-01-23 08:22:15] >> (20170516000000)
[2024-01-23 08:22:15] >> (20170517091309)
[2024-01-23 08:22:15] >> (20170607150241)
[2024-01-23 08:22:15] >> (20170607155015)
[2024-01-23 08:22:15] >> (20170621081811)
[2024-01-23 08:22:15] >> (20170621082116)
[2024-01-23 08:22:15] >> (20170628191405)
[2024-01-23 08:22:15] >> (20170725155053)
[2024-01-23 08:22:15] >> (20170728110351)
[2024-01-23 08:22:15] >> (20170810120452)
[2024-01-23 08:22:15] >> (20170829081048)
[2024-01-23 08:22:15] >> (20170906161951)
[2024-01-23 08:22:15] >> (20170923014659)
[2024-01-23 08:22:15] >> (20170926105027)
[2024-01-23 08:22:15] >> (20170928132035)
[2024-01-23 08:22:15] >> (20171024153459)
[2024-01-23 08:22:15] >> (20171027134920)
[2024-01-23 08:22:15] >> (20171031164327)
[2024-01-23 08:22:15] >> (20171208130606)
[2024-01-23 08:22:15] >> (20180105090000)
[2024-01-23 08:22:15] >> (20180117141613)
[2024-01-23 08:22:15] >> (20180201184933)
[2024-01-23 08:22:15] >> (20180220212444)
[2024-01-23 08:22:15] >> (20180329185634)
[2024-01-23 08:22:15] >> (20180329200949)
[2024-01-23 08:22:15] >> (20180404185715)
[2024-01-23 08:22:15] >> (20180406233027)
[2024-01-23 08:22:15] >> (20180410101632)
[2024-01-23 08:22:15] >> (20180419145934)
[2024-01-23 08:22:15] >> (20180507150753)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:08 (20180508202930)
[2024-01-23 08:22:15] >> (20180515082957)
[2024-01-23 08:22:15] >> (20180516201301)
[2024-01-23 08:22:15] >> (20180516211256)
[2024-01-23 08:22:15] >> (20180604000000)
[2024-01-23 08:22:15] >> (20180622152140)
[2024-01-23 08:22:15] >> (20180625222654)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:09 (20180702014364)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:09 (20180702014365)
[2024-01-23 08:22:15] >> (20180702145735)
[2024-01-23 08:22:15] >> (20180722100755)
[2024-01-23 08:22:15] >> (20180730070549)
[2024-01-23 08:22:15] >> (20180801081855)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:09 (20180821144204)
[2024-01-23 08:22:15] >> (20180829202029)
[2024-01-23 08:22:15] >> (20180830004341)
[2024-01-23 08:22:15] >> (20180906165938)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:09 (20180921144421)
[2024-01-23 08:22:15] >> (20180924120832)
[2024-01-23 08:22:15] >> (20180924120833)
[2024-01-23 08:22:15] >> (20180924120834)
[2024-01-23 08:22:15] >> (20181004211855)
[2024-01-23 08:22:15] >> (20181008234543)
[2024-01-23 08:22:15] >> (20181011150000)
[2024-01-23 08:22:15] >> (20181017154600)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:26 (20181111095447)
[2024-01-23 08:22:15] >> (20181128122944)
[2024-01-23 08:22:15] >> (20181129105419)
[2024-01-23 08:22:15] >> (20181204000000)
[2024-01-23 08:22:15] >> (20190704154940)
[2024-01-23 08:22:15] >> (20190715065013)
[2024-01-23 08:22:15] >> (20190724110039)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:10 (20191017140848)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:26 (20191126093923)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:10 (20191206113956)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:28 (20191219155630)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:30 (20200211095409)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:32 (20200212141530)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:33 (20200220172041)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:35 (20200227110431)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:36 (20200302164801)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:11 (20200331160919)
[2024-01-23 08:22:15] >> 2022-02-02 13:10:02 (20200409102100)
[2024-01-23 08:22:15] >> 2020-07-28 07:18:38 (20200422144300)
[2024-01-23 08:22:15] >> 2022-02-02 13:10:02 (20200507122854)
[2024-01-23 08:22:15] >> 2022-02-02 13:10:03 (20200729170800)
[2024-01-23 08:22:15] >> 2020-10-28 10:21:26 (20200805185714)
[2024-01-23 08:22:15] >> 2021-11-15 11:39:47 (20200810153131)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:10 (20200812180900)
[2024-01-23 08:22:15] >> 2020-10-28 10:21:26 (20200815153711)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:11 (20200924080139)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:11 (20201015084627)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:11 (20201102120710)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:11 (20201102133546)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:12 (20201105120328)
[2024-01-23 08:22:15] >> 2022-02-02 13:10:03 (20201120122846)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:10 (20201123070813)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:10 (20201125155904)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:12 (20201130201631)
[2024-01-23 08:22:15] >> 2021-11-15 11:39:47 (20201207114926)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:12 (20201207140911)
[2024-01-23 08:22:15] >> 2021-11-15 11:39:47 (20201228041109)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:13 (20210104171005)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:10 (20210203143600)
[2024-01-23 08:22:15] >> 2021-06-07 07:13:13 (20210223174702)
[2024-01-23 08:22:15] >> 2021-11-15 11:39:47 (20210502162314)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:10 (20210520100503)
[2024-01-23 08:22:15] >> 2021-11-15 11:39:47 (20210609191822)
[2024-01-23 08:22:15] >> 2021-11-15 11:39:47 (20210614151138)
[2024-01-23 08:22:15] >> 2022-02-02 13:10:03 (20210623071326)
[2024-01-23 08:22:15] >> 2022-03-01 13:06:10 (20220111202917)
[2024-01-23 08:22:15] >> 2024-01-19 07:53:22 (20230519081315)
[2024-01-23 08:22:15] >> 2024-01-19 07:53:22 (20230522115137)
[2024-01-23 08:22:15] [notice] Migrating up to Mautic\Migrations\Versionzz20230929183000
[2024-01-23 08:22:15] [notice] Table contact_export_scheduler already exists
[2024-01-23 08:22:15] [notice] Migration Mautic\Migrations\Version20220429091934 skipped during Pre-Checks. Reason: "Schema includes this migration"
[2024-01-23 08:22:15] [notice] Migration Mautic\Migrations\Version20221128145933 skipped during Pre-Checks. Reason: "Schema includes this migration"
[2024-01-23 08:22:15] [notice] Table plugin_citrix_events already exists
[2024-01-23 08:22:15] [notice] Migration Mautic\Migrations\Version20230522141144 skipped during Pre-Checks. Reason: "Schema includes this migration"
[2024-01-23 08:22:15] [notice] Migration is not required.
[2024-01-23 08:22:15] [notice] Migration Mautic\Migrations\Version20230606111852 skipped during Pre-Checks. Reason: "Schema includes this migration"
[2024-01-23 08:22:15] [notice] Table point_groups already exists
[2024-01-23 08:22:15] [notice] Table point_group_contact_score already exists
[2024-01-23 08:22:15] [notice] Column points.group_id already exists
[2024-01-23 08:22:15] [notice] Column point_triggers.group_id already exists
[2024-01-23 08:22:15] [notice] Column lead_points_change_log.group_id already exists
[error] Migration Mautic\Migrations\Version20230621074925 failed during Execution. Error: "An exception occurred while executing a query: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists"

In ExceptionConverter.php line 45:

An exception occurred while executing a query: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists

In Exception.php line 28:

SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists

In Connection.php line 71:

SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists

doctrine:migrations:migrate [--write-sql [WRITE-SQL]] [--dry-run] [--query-time] [--allow-no-migration] [--all-or-nothing [ALL-OR-NOTHING]] [--configuration CONFIGURATION] [--em EM] [--conn CONN] [--] []

Only the 20230621074925 --down

/php -d memory_limit=4096M bin/console doctrine:migration:execute 'Mautic\Migrations\Version20230621074925' --down

WARNING! You are about to execute a migration in database "mauticbeta" that could result in schema changes and data loss. Are you sure you wish to continue? (yes/no) [yes]:

[notice] Executing Mautic\Migrations\Version20230621074925 down
[notice] finished in 356.5ms, used 58.5M memory, 1 migrations executed, 5 sql queries

Only the 20230621074925 --up

php -d memory_limit=4096M bin/console doctrine:migration:execute 'Mautic\Migrations\Version20230621074925' --up

WARNING! You are about to execute a migration in database "mauticbeta" that could result in schema changes and data loss. Are you sure you wish to continue? (yes/no) [yes]:

[notice] Executing Mautic\Migrations\Version20230621074925 up
[error] Migration Mautic\Migrations\Version20230621074925 failed during Execution. Error: "An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1005 Can't create table mauticbeta.point_group_contact_score (errno: 150 "Foreign key constraint is incorrectly formed")"

In ExceptionConverter.php line 117:

An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1005 Can't create table mauticbeta.point_group_contact_score (errno: 150 "Foreign key constraint is incorrectly formed")

In Exception.php line 28:

SQLSTATE[HY000]: General error: 1005 Can't create table mauticbeta.point_group_contact_score (errno: 150 "Foreign key constraint is incorrectly formed")

In Connection.php line 71:

SQLSTATE[HY000]: General error: 1005 Can't create table mauticbeta.point_group_contact_score (errno: 150 "Foreign key constraint is incorrectly formed")

doctrine:migrations:execute [--write-sql [WRITE-SQL]] [--dry-run] [--up] [--down] [--query-time] [--configuration CONFIGURATION] [--em EM] [--conn CONN] [--] ...

Again only the 20230621074925 --up

php -d memory_limit=4096M bin/console doctrine:migration:execute 'Mautic\Migrations\Version20230621074925' --up

WARNING! You are about to execute a migration in database "mauticbeta" that could result in schema changes and data loss. Are you sure you wish to continue? (yes/no) [yes]:

[notice] Executing Mautic\Migrations\Version20230621074925 up
[notice] Table point_groups already exists
[notice] Table point_group_contact_score already exists
[notice] Column points.group_id already exists
[notice] Column point_triggers.group_id already exists
[notice] Column lead_points_change_log.group_id already exists
[error] Migration Mautic\Migrations\Version20230621074925 failed during Execution. Error: "An exception occurred while executing a query: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists"

In ExceptionConverter.php line 45:

An exception occurred while executing a query: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists

In Exception.php line 28:

SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists

In Connection.php line 71:

SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'point_groups' already exists

doctrine:migrations:execute [--write-sql [WRITE-SQL]] [--dry-run] [--up] [--down] [--query-time] [--configuration CONFIGURATION] [--em EM] [--conn CONN] [--] ...

Site runs:

mautic5_s_sysinfo
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by patrykgruszka** [comment]
@technik-kvz, @christopheg ready to test: #13284

**Comment by patrykgruszka** [comment]
Resolved by #13284
